dask/dataframe/core.py

```diff
@@ -2607,18 +2607,8 @@ def clip_upper(self, threshold):
     @derived_from(pd.DataFrame)
     def squeeze(self, axis=None):
         if axis in [None, 1]:
-            squeezed_series = self[self.columns[0]]
-
             if len(self.columns) == 1:
-                if len(self.index) == 1:
-                    # dask dataframe with only one element should return scalar
-                    dask_arr = squeezed_series.values
-                    # set chunks so we can index for first and only element
-                    dask_arr._chunks = ((1, ),)
-                    return dask_arr[0]
-                else:
-                    return squeezed_series
-
+                return self[self.columns[0]]
             else:
                 return self
```

Comment:
We should probably raise an informative `NotImplementedError` here instead so that we communicate our inability to perform this operation to the user.

Reply:
I'm not sure I see what you mean, do you mean in the case there is more than 1 column in the dataframe? The pandas implementation silently returns the same dataframe I think.

Reply:
I mean the `else` clause of the `if axis in [None, 1]` case.  Currently if the user explicitly specifies `axis=0` or `axis=2` then they get back the original dataframe without warning that this may not be what they expect.  On first glance I believe that we should raise a NotImplementedError in this case.

Reply:
I think the `else` clause you are referring to returns the original dataframe in the specific (default) case where `axis=1` or `axis=None`, but the dataframe contains more than one column so the squeeze operation doesn't do anything.

I think I have implemented a `NotImplementedError` for the case `axis=0` in [line 2615](https://github.com/dask/dask/pull/3366/files#diff-492da7893fa5fa6ad3a6ef6cdef985f2R2615) where a user might expect a `Series` for a dataframe containing a single row but multiple columns, and a `ValueError` for cases where where `axis>1` in [line 2619](https://github.com/dask/dask/pull/3366/files#diff-492da7893fa5fa6ad3a6ef6cdef985f2R2619)?

Reply:
Ah indeed.  My apologies.  I must have been looking at an older load of the diff page.  This PR looks great to me.  Merging on passed tests.

Reply:
No problem, thanks for your help and guidance. Hope I can contribute more in the future!